geoh5py/objects/block_model.py

```diff
@@ -67,210 +90,104 @@ def centroids(self) -> np.ndarray | None:
-            return self.u_cell_delimiters[1:] - self.u_cell_delimiters[:-1]
-        return None
+        return self.u_cell_delimiters[1:] - self.u_cell_delimiters[:-1]
 
     @property
-    def v_cell_delimiters(self) -> np.ndarray | None:
+    def v_cell_delimiters(self) -> np.ndarray:
         """
-        :obj:`numpy.array` of :obj:`float`:
         Nodal offsets along the v-axis relative to the origin.
         """
-        if (getattr(self, "_v_cell_delimiters", None) is None) and self.on_file:
-            self._v_cell_delimiters = self.workspace.fetch_array_attribute(
-                self, "v_cell_delimiters"
-            )
-
         return self._v_cell_delimiters
 
-    @v_cell_delimiters.setter
-    def v_cell_delimiters(self, value):
-        if value is not None:
-            value = np.r_[value]
-            self._centroids = None
-            self._v_cell_delimiters = value.astype(float)
-            self.workspace.update_attribute(self, "v_cell_delimiters")
-
     @property
-    def v_cells(self) -> np.ndarray | None:
+    def v_cells(self) -> np.ndarray:
         """
-        :obj:`numpy.array` of :obj:`float`,
-        shape (:obj:`~geoh5py.objects.block_model.BlockModel.shape` [1], ):
         Cell size along the v-axis.
         """
-        if self.v_cell_delimiters is not None:
-            return self.v_cell_delimiters[1:] - self.v_cell_delimiters[:-1]
-        return None
+        return self.v_cell_delimiters[1:] - self.v_cell_delimiters[:-1]
 
     @property
-    def z_cell_delimiters(self) -> np.ndarray | None:
+    def z_cell_delimiters(self) -> np.ndarray:
         """
-        :obj:`numpy.array` of :obj:`float`:
-        Nodal offsets along the z-axis relative to the origin (positive up).
+        Nodal offsets along the u-axis relative to the origin.
         """
-        if (getattr(self, "_z_cell_delimiters", None) is None) and self.on_file:
-            self._z_cell_delimiters = self.workspace.fetch_array_attribute(
-                self, "z_cell_delimiters"
-            )
-
         return self._z_cell_delimiters
 
-    @z_cell_delimiters.setter
-    def z_cell_delimiters(self, value):
-        if value is not None:
-            value = np.r_[value]
-            self._centroids = None
-            self._z_cell_delimiters = value.astype(float)
-            self.workspace.update_attribute(self, "z_cell_delimiters")
-
     @property
-    def z_cells(self) -> np.ndarray | None:
+    def z_cells(self) -> np.ndarray:
         """
-        :obj:`numpy.array` of :obj:`float`,
-        shape (:obj:`~geoh5py.objects.block_model.BlockModel.shape` [2], ):
         Cell size along the z-axis
         """
-        if self.z_cell_delimiters is not None:
-            return self.z_cell_delimiters[1:] - self.z_cell_delimiters[:-1]
-        return None
+        return self.z_cell_delimiters[1:] - self.z_cell_delimiters[:-1]
+
+    @staticmethod
+    def validate_cell_delimiters(value: np.ndarray, axis: str) -> np.ndarray:
```

Comment:
is it enought in terms of validation?
Cells shouldn't be equals in x, y, z for example?

Reply:
Nope, three independent vectors